src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -146,7 +146,7 @@ var ReactDOMInput = {
 
     var defaultValue = props.defaultValue;
     inst._wrapperState = {
-      initialChecked: props.checked != null ? props.checked : props.defaultChecked,
+      initialChecked: props.checked !== undefined ? props.checked : props.defaultChecked,
       initialValue: props.value != null ? props.value : defaultValue,
```

Comment:
If accepted, wouldn't this introduce a new inconsistency between controlled text inputs and checkboxes?